atomicapp/providers/marathon.py

```diff
@@ -0,0 +1,122 @@
+            self.marathon_api = urlparse.urljoin(self.marathon_api, "v2/")
+
+        logger.debug("marathon_api = %s", self.marathon_api)
+        self._process_artifacts()
+
+    def deploy(self):
+        """ Deploy containers
+        """
+        for artifact in self.marathon_artifacts:
+            url = urlparse.urljoin(self.marathon_api, "apps/")
+
+            if self.dryrun:
+                logger.info("DRY-RUN: %s", url)
+                continue
+
+            logger.debug("Deploying appid: %s", artifact["id"])
+            res = requests.post(url, json=artifact)
+            if res.status_code == 201:
+                logger.info(
+                    "Marathon app %s sucessfully deployed.",
+                    artifact["id"])
+            else:
+                msg = "Error deploying app: %s, Marathon API response %s - %s" % (
+                    artifact["id"], res.status_code, res.text)
+                logger.error(msg)
+                raise ProviderFailedException(msg)
+
+    def undeploy(self):
+        """ Undeploy containers
+        """
+        for artifact in self.marathon_artifacts:
+            url = urlparse.urljoin(
+                self.marathon_api,
+                "apps/%s" %
+                artifact["id"])
+
+            if self.dryrun:
+                logger.info("DRY-RUN: %s", url)
+                continue
+
+            logger.debug("Deleting appid: %s", artifact["id"])
+            res = requests.delete(url, json=artifact)
+            if res.status_code == 200:
+                logger.info(
+                    "Marathon app %s sucessfully deleted.",
+                    artifact["id"])
+            else:
+                msg = "Error deleting app: %s, Marathon API response %s - %s" % (
+                    artifact["id"], res.status_code, res.text)
+                logger.error(msg)
+                raise ProviderFailedException(msg)
+
+    def _process_artifacts(self):
+        """
+        Parse and validate Marathon artifacts and save them to self.marathon_artifacts
+        """
+        for artifact in self.artifacts:
+            logger.debug("Procesesing artifact: %s", artifact)
+            data = None
+            with open(os.path.join(self.path, artifact), "r") as fp:
+                try:
+                    data = anymarkup.parse(fp)
+                    logger.debug("Parsed artifact %s", data)
+                    # every marathon app has to have id
+                    # 'id' key  is also used for showing messages
+                    if "id" not in data.keys():
+                        raise ProviderFailedException(
+                            "Error processing %s artifact. There is no id" % artifact)
+                except Exception:
+                    msg = "Error processing %s artifcat. Error:" % os.path.join(
+                        self.path, artifact)
+                    printErrorStatus(msg)
+                    raise
```

Comment:
"empty' raise re-rise previews exeption ;-) I took this from kubernetes provider.

Besides `ProviderFailedException`  only exception that can be raised in that block is `AnyMarkupError`. This could be handled in better way, for sure.